src/KafkaKeyValue.ts

```diff
@@ -207,53 +231,53 @@ export default class KafkaKeyValue {
     this.fetchImpl = getFetchImpl(config);
     this.logger = getLogger({ name: `kkv:${this.getCacheName()}` });
 
-    updateEvents.on('update', async (requestBody: { v: number, offsets: { [partition: string]: number }, topic: string, updates: { [key: string]: {} } }) => {
-      if (requestBody.v !== 1) throw new Error(`Unknown kkv onupdate protocol ${requestBody.v}!`);
+    updateEvents.on('update', this.updateListener.bind(this));
+  }
 
-      const {
-        topic, offsets, updates
-      } = requestBody;
+  async updateListener(requestBody: UpdateRequestBody) {
+    if (requestBody.v !== 1) throw new Error(`Unknown kkv onupdate protocol ${requestBody.v}!`);
 
-      const expectedTopic = this.topic;
-      this.logger.trace({ topic, expectedTopic }, 'Matching update event against expected topic');
-      if (topic !== expectedTopic) {
-        this.logger.trace({ topic, expectedTopic }, 'Update event ignored due to topic mismatch. Business as usual.');
-        return;
-      }
+    const {
+      topic, offsets, updates
+    } = requestBody;
 
-      const highestOffset: number = Object.values(offsets).reduce((memo, offset) => {
-        return Math.max(memo, offset);
-      }, -1);
+    const expectedTopic = this.topic;
+    this.logger.trace({ topic, expectedTopic }, 'Matching update event against expected topic');
```

Comment:
Why isn't this in an `else`?

Reply:
Clarification: I misunderstood "Matching update" as "update matches expected topic". Even if that's not the intent, maybe it's better to log in an else below, than to always log "Matching ...". But now it's a nit.